modules/metastorage-common/src/main/java/org/apache/ignite/metastorage/common/Condition.java

```diff
@@ -0,0 +1,357 @@
+         * is less than the given revision.
+         *
+         * @param rev The revision.
+         * @return The condition of type {@link Type#LESS}.
+         * @throws IllegalStateException In case when the condition is already defined.
+         */
+        public Condition lt(long rev) {
+            validate(type);
+
+            this.type = Type.LESS;
+            this.rev = rev;
+
+            return new Condition(this);
+        }
+
+        /**
+         * Produces the condition of type {@link Type#LESS_OR_EQUAL}. This condition tests that target entry revision
+         * is less than or equal to the given revision.
+         *
+         * @param rev The revision.
+         * @return The condition of type {@link Type#LESS_OR_EQUAL}.
+         * @throws IllegalStateException In case when the condition is already defined.
+         */
+        public Condition le(long rev) {
+            validate(type);
+
+            this.type = Type.LESS_OR_EQUAL;
+            this.rev = rev;
+
+            return new Condition(this);
+        }
+
+        /** {@inheritDoc} */
+        @Override public boolean test(Entry e) {
+            int res = Long.compare(e.revision(), rev);
+
+            return type.test(res);
+        }
+
+        /**
+         * Defines possible condition types which can be applied to the revision.
+         */
+        enum Type {
+            /** Equality condition type. */
+            EQUAL {
+                @Override public boolean test(long res) {
+                    return res == 0;
+                }
+            },
+
+            /** Inequality condition type. */
+            NOT_EQUAL {
+                @Override public boolean test(long res) {
+                    return res != 0;
+                }
+            },
+
+            /** Greater than condition type. */
+            GREATER {
+                @Override public boolean test(long res) {
+                    return res > 0;
+                }
+            },
+
+            /** Less than condition type. */
+            LESS {
+                @Override public boolean test(long res) {
+                    return res < 0;
+                }
+            },
+
+            /** Less than or equal to condition type. */
+            LESS_OR_EQUAL {
+                @Override public boolean test(long res) {
+                    return res <= 0;
+                }
+            },
+
+            /** Greater than or equal to condition type. */
+            GREATER_OR_EQUAL {
+                @Override public boolean test(long res) {
+                    return res >= 0;
+                }
+            };
+
+            /**
+             * Interprets comparison result.
+             *
+             * @param res The result of comparison.
+             * @return The interpretation of the comparison result.
+             */
+            public abstract boolean test(long res);
```

Comment:
Will be 'eval' more suitable method name?

Reply:
`eval` assumes that any type of result could be returned, while `test` is the common name for predicate-like methods. See for example `java.util.function.Predicate` interface.